File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/decorator/Finery.java
package jv.com.tencent.cain.design_pattern.decorator;

/**
 * @author cainjiang
 * @date 2019-03-31
 * 装饰者的父类，持有一个被装饰者的引用
 */
public class Finery extends Person {
    private Person compenent;

    protected void decorate(Person compenent) {
        this.compenent = compenent;
    }

    @Override
    protected void show() {
        if (compenent != null) {
            compenent.show();
        }
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/decorator/Finery.kt
package kt.com.tencent.cain.desigin_pattern.decorator

/**
 * @author cainjiang
 * @date 2019-03-31
 * 装饰者的父类,持有一个被装饰者的引用
 */
open class Finery(private var compenet: Person? = null) : Person() {
    fun decorate(compenet: Person?) {
        this.compenet = compenet
    }

    override fun show() {
        compenet?.show()
    }
}